### ISSUE TITLE
Credit mining is incompatible with older versions of libtorrent

### ISSUE THREAD

**Issue by devos50** [Tribler/tribler#2676]
I'm getting this error when running the credit mining tests on a computer with libtorrent 0.16.18:

```
test_logging (Tribler.Test.Core.CreditMining.test_creditmining.TestBoostingManagerUtilities) ... ERROR
ERROR

======================================================================
ERROR: test_logging (Tribler.Test.Core.CreditMining.test_creditmining.TestBoostingManagerUtilities)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/martijn/tribler/Tribler/Test/test_as_server.py", line 64, in check
    result = fun(*argv, **kwargs)
  File "/home/martijn/tribler/Tribler/Test/Core/CreditMining/test_creditmining.py", line 358, in test_logging
    boost_man.log_statistics()
  File "/home/martijn/tribler/Tribler/Policies/BoostingManager.py", line 458, in log_statistics
    if lt.__version__ == '1.0.9.0':
AttributeError: 'module' object has no attribute '__version__'
-------------------- >> begin captured logging << --------------------
BoostingManager: DEBUG: Status for 6161616161616161616161616161616161616161 : 0 0 | ul_lim : 12, max_ul 13, maxcon 14
--------------------- >> end captured logging << ---------------------
```

Note that `__version__` is not available in older versions of libtorrent.

**Comment by ardhipoetra** [comment]
I tried to compile libtorrent  0.16.18, but it gives me error `undefined symbol: _ZTIN10libtorrent7torrentE` upon launching.

As it's only used for logging, I can wrap it in try-except Attribute error. If the version is unknown, we proceed with not logs anything. Does this approach acceptable?

**Comment by devos50** [comment]
Older versions of libtorrent require `libtorrent.version` to get the version. We should create a method in `LibtorrentMgr` that returns the version string, independent of which version of libtorrent is being used.

I'm sure we have another instance of this somewhere in the code where I had to put a conditional check for the libtorrent version.

**Comment by ardhipoetra** [comment]
I found the conditional code that depends on libtorrent version, but didn't found anything that checks the actual version. It checks the attribute instead. Examples : 
- https://github.com/Tribler/tribler/blob/devel/Tribler/Core/Libtorrent/LibtorrentMgr.py#L302
- https://github.com/Tribler/tribler/blob/devel/Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py#L742
- https://github.com/Tribler/tribler/blob/devel/Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py#L786

**Comment by devos50** [comment]
Hmm that's weird, I thought I had I've implemented something like that to get the version number of libtorrent.

**Comment by devos50** [comment]
This is fixed in #2677
